scalafmt-cli/src/main/scala/org/scalafmt/cli/Cli.scala

```diff
@@ -106,60 +105,24 @@ object Cli {
       case DiffFiles(branch) =>
         options.gitOps.diff(branch).filter(canFormat)
     }
-
+    val excludeRegexp = options.excludeFilterRegexp
+    files.filter { f =>
+      excludeRegexp.findFirstIn(f.path).isEmpty
+    }
   }
 
-  private def getInputMethods(options: CliOptions): Seq[InputMethod] = {
+  private def getInputMethods(
+      options: CliOptions,
+      filter: Option[FilterMatcher]): Seq[InputMethod] = {
```

Comment:
Accept `Option[FilterMatcher]` for the following cases.

- `runDynamic` doesn't want to filter files on client side (scalafmt-dynamic will do that instead).
- `runScalafmt` want to filter files here as the original `getInputMethods` did.